superset-frontend/src/components/ButtonGroup/index.tsx

```diff
@@ -30,22 +31,28 @@ export default function ButtonGroup(props: ButtonGroupProps) {
       role="group"
       className={className}
       css={{
-        '& :nth-of-type(1):not(:nth-last-of-type(1))': {
+        display: 'flex',
+        '& > :nth-of-type(1):not(:nth-last-of-type(1))': {
           borderTopRightRadius: 0,
           borderBottomRightRadius: 0,
           borderRight: 0,
           marginLeft: 0,
         },
-        '& :not(:nth-of-type(1)):not(:nth-last-of-type(1))': {
+        '& > :not(:nth-of-type(1)):not(:nth-last-of-type(1))': {
           borderRadius: 0,
           borderRight: 0,
           marginLeft: 0,
         },
-        '& :nth-last-of-type(1):not(:nth-of-type(1))': {
+        '& > :nth-last-of-type(1):not(:nth-of-type(1))': {
           borderTopLeftRadius: 0,
           borderBottomLeftRadius: 0,
           marginLeft: 0,
         },
+        ...(props.expand && {
```

Comment:
Adding a new "expand" prop that makes buttons take up the available width (stretching).

Important side note that I GREATLY prefer adding props or classes to components, rather than adding overrides in their individual instances (that makes a mess).